databases/pxlib/Portfile

```diff
@@ -28,22 +27,30 @@ checksums           rmd160  1bf8b7e14c92f47058aebd1a675b34a76b720660 \
                     sha256  4d0a84a36d46a81ff8c629c4a00de3e7b6e7327d5d56c0a2ae44c1ea715ae63f \
                     size    532895
 
-depends_build       port:pkgconfig \
-                    port:intltool \
-                    port:autoconf \
-                    port:automake \
-                    port:libtool
-
-depends_lib         port:libgsf \
-                    port:libiconv
-
 # reconfigure using autogen.sh for intltool 0.51 compatibility
 
 patchfiles          patch-autogen.sh.diff \
                     patch-configure.ac.diff
 
 configure.cmd       ./autogen.sh && ./configure
 
-configure.args      --disable-silent-rules \
+depends_build-append \
+                    port:autoconf \
+                    port:automake \
+                    port:gettext \
+                    port:intltool \
+                    port:libtool \
+                    port:pkgconfig
+
+depends_lib-append  port:libgsf \
+                    port:libiconv
+
+configure.args-append \
+                    --disable-silent-rules \
                     --enable-option-checking \
                     --with-gsf=${prefix}
+
+if {[string match *clang* ${configure.compiler}]} {
```

Comment:
This will of course match all versions of clang. If all versions of clang that MacPorts might select for this port understand this flag, then that's fine.

Reply:
I have no idea how to check it. If you recommend restricting it, let’s do it.

Reply:
I think Ryan expects you to check which clang versions might not support this flag, not him to check them.

Reply:
it looks like it came into being here:

https://github.com/llvm/llvm-project/commit/d9b7dfe4a4dffed4455fbfcc1cc7b52b63895174

which matches up with clang-4.0

So -- I would say that you would add this flag on clang 16+, where it is needed . What is harder is to accurately specify which xcode clangs to add it to. You have to find that table that matches the xcode version to the llvm version, but maybe just adding it on newest xcodes would be OK, ie xcode 15 and newer.

Reply:
Thank you, Ken, I will do this.

Reply:
well, this was done just a couple of weeks ago….

https://github.com/macports/macports-ports/commit/0a843ecca89a4ac808ad37d2ce6801dd3a42ad63

Reply:
> well, this was done just a couple of weeks ago….
> 
> [0a843ec](https://github.com/macports/macports-ports/commit/0a843ecca89a4ac808ad37d2ce6801dd3a42ad63)

There is no check for a clang version there though.

Reply:
@ryandesign @kencu Given that `libsecret` builds fine on every system with Clang: https://ports.macports.org/port/libsecret/details
I guess this is good, we do not need to change anything, and the PR can be merged.